visma/solvers/polynomial/cubic.py

```diff
@@ -0,0 +1,216 @@
+    if h <= 0:
+        if h == 0 and g == 0 and f == 0:
+            # All three (real) roots exist and are equal
+            animations += [[]]
+            comments += [['Hence, three equal real roots exist.']]
+            res = cubeRoot(d/a)
+            valueX1 = [-res, 0]
+            valueX2 = [-res, 0]
+            valueX3 = [-res, 0]
+            roots.append(valueX1)
+        else:
+            # All three (real) roots exist
+            animations += [[]]
+            comments += [['Hence, three equal non-equal real roots exist.']]
+            i = (((g**2)/4) - h) ** (1./2.)
+            j = cubeRoot(i)
+            k = math.acos(-g/(2*i))
+            L = j * (-1)
+            M = math.cos(k/3)
+            N = math.sqrt(3) * (math.sin(k/3))
+            P = -(b/(3*a))
+            valueX1 = [2*j*(math.cos(k/3)) - (b/(3*a)), 0]
+            valueX2 = [L*(M + N) + P, 0]
+            valueX3 = [L*(M - N) + P, 0]
+            roots.extend([valueX1, valueX2, valueX3])
+    else:
+        # Only one (real) root exists
+        animations += [[]]
+        comments += [['Hence, one real root exists']]
+        R = -(g/2) + h ** (1./2.)
+        S = cubeRoot(R)
+        T = -(g/2) - (h ** (1./2.))
+        U = cubeRoot(T)
+        valueX1 = [(S + U) - (b/(3*a)), 0]
+
+        valueRealX2 = -(S + U)/2 - (b/(3*a))
+        valueImagX2 = (S - U)*(3 ** (1./2.))/2
+        valueX2 = [valueRealX2, valueImagX2]
+
+        valueRealX3 = -(S + U)/2 - (b/(3*a))
+        valueImagX3 = -(S - U)*(3 ** (1./2.))/2
+        valueX3 = [valueRealX3, valueImagX3]
+        roots.extend([valueX1, valueX2, valueX3])
+
+    return roots, animations, comments
+
+
+def cubicRoots(lTokens, rTokens):
+    '''Used to get roots of a cubic equation
+    This functions also translates roots {list} into final result of solution
+
+    Argument:
+        lTokens {list} -- list of LHS tokens
+        rTokens {list} -- list of RHS tokens
+
+    Returns:
+        lTokens {list} -- list of LHS tokens
+        rTokens {list} -- list of RHS tokens
+        {empty list}
+        token_string {string} -- final result stored in a string
+        animation {list} -- list of equation solving process
+        comments {list} -- list of comments in equation solving process
+    '''
+    from visma.solvers.polynomial.roots import getCoefficients
+
+    animations = []
+    comments = []
+    lTokens, rTokens, _, token_string, animNew1, commentNew1 = simplifyEquation(lTokens, rTokens)
+    animations.extend(animNew1)
+    comments.extend(commentNew1)
+    if len(rTokens) > 0:
+        lTokens, rTokens = moveRTokensToLTokens(lTokens, rTokens)
+    coeffs = getCoefficients(lTokens, rTokens, 3)
+    var = getVariables(lTokens)
+    roots, animNew2, commentNew2 = getRootsCubic(coeffs)
+    animations.extend(animNew2)
+    comments.extend(commentNew2)
+    tokens1 = []
+    expression1 = Expression()
+    expression1.coefficient = 1
+    expression1.power = 3
+    variable = Variable()
```

Comment:
```suggestion
    variable = Variable(1, var, 1)
```
Can use the same method for initializing the `Expression` and `Constant` objects at other places. The Expression's `__init__` might need a little fixing.

Reply:
Done.